barretenberg/cpp/src/barretenberg/stdlib/primitives/biggroup/biggroup.test.cpp

```diff
@@ -414,6 +414,79 @@ template <typename TestType> class stdlib_biggroup : public testing::Test {
         EXPECT_CIRCUIT_CORRECTNESS(builder);
     }
 
+    static void test_short_scalar_mul()
+    {
+        Builder builder;
+        size_t num_repetitions = 1;
+        for (size_t i = 0; i < num_repetitions; ++i) {
+            affine_element input(element::random_element());
+            // Get 128-bit scalar
+            uint256_t scalar_raw = fr::random_element();
+            scalar_raw.data[2] = 0ULL;
+            scalar_raw.data[3] = 0ULL;
+            fr scalar = fr(scalar_raw);
+            // Add skew
+            if (uint256_t(scalar).get_bit(0)) {
+                scalar -= fr(1);
+            }
+            element_ct P = element_ct::from_witness(&builder, input);
+            scalar_ct x = scalar_ct::from_witness(&builder, scalar);
+
+            // Set input tags
+            x.set_origin_tag(challenge_origin_tag);
+            P.set_origin_tag(submitted_value_origin_tag);
+
+            std::cerr << "gates before mul " << builder.get_estimated_num_finalized_gates() << std::endl;
+            element_ct c = P.template scalar_mul<128>(x);
+            std::cerr << "builder aftr mul " << builder.get_estimated_num_finalized_gates() << std::endl;
+            affine_element c_expected(element(input) * scalar);
+
+            // Check the result of the multiplication has a tag that's the union of inputs' tags
+            EXPECT_EQ(c.get_origin_tag(), first_two_merged_tag);
+            fq c_x_result(c.x.get_value().lo);
+            fq c_y_result(c.y.get_value().lo);
+
+            EXPECT_EQ(c_x_result, c_expected.x);
+
+            EXPECT_EQ(c_y_result, c_expected.y);
+        }
+
+        EXPECT_CIRCUIT_CORRECTNESS(builder);
+    }
+
+    static void test_short_scalar_mul_infinity()
+    {
+        Builder builder;
+        element input = element::infinity();
+
+        // Get 128-bit scalar
+        uint256_t scalar_raw = fr::random_element();
+        scalar_raw.data[2] = 0ULL;
+        scalar_raw.data[3] = 0ULL;
+        fr scalar = fr(scalar_raw);
+        // Add skew
+        if (uint256_t(scalar).get_bit(0)) {
+            scalar -= fr(1);
+        }
+        element_ct P = element_ct::from_witness(&builder, input);
+        scalar_ct x = scalar_ct::from_witness(&builder, scalar);
+
+        // Set input tags
+        x.set_origin_tag(challenge_origin_tag);
+        P.set_origin_tag(submitted_value_origin_tag);
+
+        std::cerr << "gates before mul " << builder.get_estimated_num_finalized_gates() << std::endl;
+        element_ct c = P.template scalar_mul<128>(x);
```

Comment:
You can check the number of gates

Reply:
And since you've created the arbitrary size scalar mul, could you create a test for every number of bits?